pkg/storage/stores/local/shipper.go

```diff
@@ -260,20 +249,14 @@ func (s *Shipper) syncLocalWithStorage(ctx context.Context) (err error) {
 
 // deleteFileFromCache removes a file from cache.
 // It takes care of locking the filesCollection, closing the boltdb file and removing the file from cache
-func (s *Shipper) deleteFileFromCache(period, uploader string, fc *filesCollection) error {
+func (s *Shipper) deleteFileFromCache(uploader string, fc *filesCollection) error {
 	fc.Lock()
 	defer fc.Unlock()
 
-	if err := fc.files[uploader].boltdb.Close(); err != nil {
-		return err
-	}
-
-	delete(fc.files, uploader)
-
-	return os.Remove(path.Join(s.cfg.CacheLocation, period, uploader))
+	return fc.cleanupFile(uploader)
 }
 
-func (s *Shipper) forEach(ctx context.Context, period string, callback func(db *bbolt.DB) error) error {
+func (s *Shipper) forEach(period string, callback func(db *bbolt.DB) error) error {
 	s.downloadedPeriodsMtx.RLock()
```

Comment:
It works in following manner:
1. Check with read-only lock whether we can find the entry.
2. If not with write lock check whether it got created by some other goroutine due to a race because multiple goroutines might be trying to acquire the write lock and create it.
3. If it is still not there create it.

Reply:
Thanks. Would you mind adding this as a comment to the code here?